pkg/odo/cli/url/create.go

```diff
@@ -92,6 +92,8 @@ func NewURLCreateOptions() *URLCreateOptions {
 
 // Complete completes URLCreateOptions after they've been Created
 func (o *URLCreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	o.DevfilePath = clicomponent.DevfilePath
```

Comment:
In `push.go` we use `po.DevfilePath = filepath.Join(po.componentContext, po.DevfilePath)` for context support, similar with `odo delete`, but I think `odo url` uses different way to do the context support as there is no `componentContext` in URLCreateOptions struct before and previously `odo url` just needs take the path from `--devfile`, the change in this PR is to let `o.DevfilePath` points to the global exported variable `DevfilePath` instead of the path from `--devfile`, and the context support should be handled implicitly in the following context functions.

I test the `odo url` context support with this PR, looks like it works for me and we don't need to explicitly handle the context support
```
odo create nodejs --context ./odo-test                                                                                                                                             ✔  17:11:42 
Experimental mode is enabled, use at your own risk

Validation
 ✓  Checking devfile compatibility [62731ns]
 ✓  Creating a devfile component from registry: Test [47635ns]
 ✓  Validating devfile component [269039ns]

Please use `odo push` command to create the component with source deployed
    ~/wang  odo url create test --context ./odo-test                                                                                                                                           ✔  17:12:03 
 ✓  URL test created for component: nodejs

To apply the URL configuration changes, please use `odo push`
    ~/wang  cd odo-test                                                                                                                                                                        ✔  17:12:15 
    ~/wang/odo-test  cat ./.odo/env/env.yaml                                                                                                                                                   ✔  17:12:20 
ComponentSettings:
  Name: nodejs
  Namespace: default
  Url:
  - Name: test
    Port: 3000
    Kind: route
```